code/modules/mob/living/silicon/robot/robot.dm

```diff
@@ -978,6 +987,36 @@
 	set category = "Robot Commands"
 	set_autosay()
 
+/mob/living/silicon/robot/proc/control_headlamp()
+	if(stat)
+		src << "<span class='danger'>This function is currently offline.</span>"
+		return
+
+	if(lamp_intensity < lamp_max) //Lamp is set lower than the max.
+		lamp_intensity += 2
+		src << "Headlamp power set to Level [lamp_intensity/2]."
+	else //Greater than or equal to the max, so turn it off instead.
+		lamp_intensity = 0
+		src << "Headlamp disabled."
+
+	update_headlamp()
+
+/mob/living/silicon/robot/proc/update_headlamp(var/turn_off = 0)
+	SetLuminosity(0)
+
+	if(lamp_intensity && (turn_off || stat))
+		src << "<span class='danger'>Your headlamp has been deactivated.</span>"
+		lamp_intensity = 0
+		if(client)
+			var/obj/screen/robot/lamp/button = locate(/obj/screen/robot/lamp) in client.screen

```

Comment:
I have seen other maintainers say not to use for() loops when hunting for only one element in something. Has this precedent changed?